integrationtests/src/test/java/io/kroxylicious/proxy/ResilienceIT.java

```diff
@@ -71,9 +71,10 @@ private static void testConsumerCanSurviveKroxyliciousRestart(ConfigurationBuild
                 ConsumerConfig.GROUP_ID_CONFIG, "mygroup",
                 ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"));
 
+        Producer<String, String> producer;
         Consumer<String, String> consumer;
-        try (var tester = kroxyliciousTester(builder);
-                var producer = tester.producer(producerConfig)) {
+        try (var tester = kroxyliciousTester(builder)) {
+            producer = tester.producer(producerConfig);
```

Comment:
No. You've leaked a producer with this change as #close is no longer being called.  The code was deliberate in its scoping of the producer/consumer. 
- producer being scoped to the first kroxy instance
- consumer being used by both (so we prove it reconnects)

Reply:
Indeed, missed that earlier. I think I was trying to make it consistent with the following test (`testProducerCanSurviveARestart`) but wasn't thinking at all about the context of the test and why it was laid out that way to begin with. I've fixed this now.